agenta-backend/agenta_backend/models/db_models.py

```diff
@@ -1,303 +1,396 @@
-
-
-class EvaluationDB(Document):
-    app: Link[AppDB]
-    user: Link[UserDB]
-    status: Result
-    testset: Link[TestSetDB]
-    variant: PydanticObjectId
-    variant_revision: PydanticObjectId
-    evaluators_configs: List[PydanticObjectId]
-    aggregated_results: List[AggregatedResult]
-    average_cost: Optional[Result] = None
-    total_cost: Optional[Result] = None
-    average_latency: Optional[Result] = None
-    created_at: datetime = Field(default=datetime.now(timezone.utc))
-    updated_at: datetime = Field(default=datetime.now(timezone.utc))
-
-    class Settings:
-        name = "new_evaluations"
-
-
-class CorrectAnswer(BaseModel):
-    key: str
-    value: str
-
-
-class EvaluationScenarioDB(Document):
-    user: Link[UserDB]
-    evaluation: Link[EvaluationDB]
-    variant_id: PydanticObjectId
-    inputs: List[EvaluationScenarioInputDB]
-    outputs: List[EvaluationScenarioOutputDB]
-    correct_answers: Optional[List[CorrectAnswer]]
-    is_pinned: Optional[bool]
-    note: Optional[str]
-    evaluators_configs: List[PydanticObjectId]
-    results: List[EvaluationScenarioResult]
-    latency: Optional[int] = None
-    cost: Optional[int] = None
-    created_at: datetime = Field(default=datetime.now(timezone.utc))
-    updated_at: datetime = Field(default=datetime.now(timezone.utc))
-
-    class Settings:
-        name = "new_evaluation_scenarios"
+from typing import Any, Dict, List, Optional
+from sqlalchemy import (
+    Column,
+    String,
+    Integer,
+    DateTime,
+    Boolean,
+    ForeignKey,
+    JSON,
+    Text,
+    Float,
+)
+from sqlalchemy.orm import relationship, declarative_base
+
+Base = declarative_base()
+
+
+class UserDB(Base):
+    id = Column(Integer, primary_key=True, autoincrement=True)
+    uid = Column(String, unique=True, index=True, default="0")
+    username = Column(String, default="agenta")
+    email = Column(String, unique=True, default="demo@agenta.ai")
+    created_at = Column(
+        DateTime(timezone=True), default=lambda: datetime.now(timezone.utc)
+    )
+    updated_at = Column(
+        DateTime(timezone=True), default=lambda: datetime.now(timezone.utc)
+    )
+
+    __tablename__ = "users"
+
+
+# TODO: Rename ImageDB to DockerImageDB ?
+class ImageDB(Base):
+    id = Column(Integer, primary_key=True, autoincrement=True)
+    type = Column(String, default="image")
+    template_uri = Column(String)
+    docker_id = Column(String, index=True)
+    tags = Column(String)
+    deletable = Column(Boolean, default=True)
+    user_id = Column(Integer, ForeignKey("users.id"))
+    user = relationship("UserDB")
+    created_at = Column(
+        DateTime(timezone=True), default=lambda: datetime.now(timezone.utc)
+    )
+    updated_at = Column(
+        DateTime(timezone=True), default=lambda: datetime.now(timezone.utc)
+    )
+
+    __tablename__ = "docker_images"
+
+
+class AppDB(Base):
+    id = Column(Integer, primary_key=True, autoincrement=True)
```

Comment:
would we create new ids here in the migration or reuse our ids, if we reuse, would autoincrement work (cc @aybruhm )